utils/utils.go

```diff
@@ -60,6 +61,21 @@ func Unzip(archive, target string) error {
 	return nil
 }
 
+// TODO using cmd := exec.CommandContext(ctx,...
```

Comment:
Pending... nice to have an integration with exec.CommandContext(ctx

Reply:
I actually would start using CommandContext here right away and propagate all the way up the call chain the ctx parameter. Whenever you reach a method that does not allow having its interface easily modified, use `context.Background()` instead of a parameter for it.